### ISSUE TITLE
Lost daily streak every month change

### ISSUE THREAD

**Issue by Kamizun** [nuclearsilo583/zephyrus-store-preview-new-syntax#125]
Last month change I noticed this error but I was not sure it was the plugin fault.
This month change I made sure to try again and noticed I lost my streak again.
I decided to investigate and found that the plugin saves my last daily as "20221031" and then subtracts with the current time "20221101"
Well 20221031-20221101 = 70 so I lost my "70" day streak. The same thing happened last month 20220930-20221001=71

I screenshotted both these when they happened:

![Screenshot_1](https://user-images.githubusercontent.com/20317669/199218314-db5e6ff9-0463-4a27-8f76-f0f27cf01f90.png)
![Screenshot_2](https://user-images.githubusercontent.com/20317669/199218316-e367628e-7dc9-404d-9c83-9f37e54d8ef9.png)

**Comment by azalty** [comment]
Yes indeed something is clearly wrong about how time is stored and compared
